lua/entities/gmod_wire_expression2/core/timer.lua

```diff
@@ -113,10 +200,178 @@ e2function void stopAllTimers()
 		self.prf = self.prf + 5
 		RemoveTimer(self,name)
 	end
+
+	for k, _ in pairs(luaTimers[self.entity:EntIndex()] or {}) do
+		self.prf = self.prf + 5
+		luaTimerRemove(self, k)
+	end
 end
 
 /******************************************************************************/
+-- Lambda timers
+
+__e2setcost(10)
+e2function void timer(string name, number delay, number repetitions, function callback)
+	luaTimerCreate(self, name, delay, repetitions, callback)
+end
+
+e2function string timer(number delay, number repetitions, function callback)
+	local name = "simpletimer_"..luaTimerGetNextIncrementalKey(self)
+	luaTimerCreate(self, name, delay, repetitions, callback)
+	return name
+end
 
+e2function string timer(number delay, function callback)
+	local name = "simpletimer_"..luaTimerGetNextIncrementalKey(self)
+	luaTimerCreate(self, name, delay, 1, callback)
+	return name
+end
+
+e2function void timer(string name, number delay, function callback)
+	luaTimerCreate(self, name, delay, 1, callback)
+end
+
+__e2setcost(5)
+e2function void timerSetDelay(string name, number delay)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", nil)
+	end
+
+	local entIndex = self.entity:EntIndex()
+	luaTimers[entIndex][name].delay = delay
+	timer.Adjust(luaTimerGetInternalName(entIndex, name), delay, 0)
+end
+
+e2function number timerSetReps(string name, number repetitions)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", nil)
+	end
+
+	local entIndex = self.entity:EntIndex()
+	luaTimers[entIndex][name].repetitions = repetitions
+	timer.Adjust(luaTimerGetInternalName(entIndex, name), luaTimers[entIndex][name].delay, repetitions)
+end
+
+e2function void timerAdjust(string name, number delay, number repetitions)
+	if not luaTimerExists(self, name) then
+		return self:throw("Timer with name " .. name .. " does not exist", nil)
+	end
+
+	local entIndex = self.entity:EntIndex()
+	luaTimers[entIndex][name].delay = delay
+	luaTimers[entIndex][name].repetitions = repetitions
+	timer.Adjust(luaTimerGetInternalName(entIndex, name), delay, repetitions)
+end
+
+-- __e2setcost(20)
+-- e2function void timerSetCallback(string name, function callback)
+-- 	if not luaTimerExists(self, name) then
+-- 		return self:throw("Timer with name " .. name .. " does not exist", nil)
+-- 	end
+
+-- 	local int_name = luaTimerGetInternalName(self.entity:EntIndex(), name)
+-- 	local luaTimer = table.Copy(luaTimers[self.entity:EntIndex()][int_name])
+-- 	luaTimerRemove(self, name)
+
+-- 	luaTimer['callback'] = callback
+-- 	luaTimerCreate(self, name, luaTimer.delay, luaTimer.repetitions, callback)
+-- end
```

Comment:
Is this commented for some reason?

Reply:
Was not sure either we want it or not. Ig not